scripts/delete-bucket

```diff
@@ -41,10 +41,16 @@ def main():
     paginator = s3_client.get_paginator("list_objects_v2")
     try:
         for page in paginator.paginate(Bucket=bucket_name):
-            s3_client.delete_objects(
-                Bucket=bucket_name,
-                Delete={"Objects": [{"Key": obj["Key"]} for obj in page["Contents"]]},
-            )
+            try:
+                s3_client.delete_objects(
+                    Bucket=bucket_name,
+                    Delete={"Objects": [{"Key": obj["Key"]} for obj in page["Contents"]]},
+                )
+            except KeyError as e:
+                # Empty buckets do not have a Contents key in the reply.
+                [missing_key] = e.args
+                if missing_key != "Contents":
+                    raise e
```

Comment:
Ça ne marcherais pas de tester que [`page["KeyCount"] > 0` ](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/list_objects_v2.html) avant le `.delete_objects()` ?
Ça me semble un peu moins bourrin :grin:.

Reply:
Il n’y a rien qui dit que `KeyCount` sera toujours présent, tout comme rien ne dit que `Contents` sera potentiellement absent... Autant vérifier la présence de ce qu’on utilise ?

Reply:
Après, on peut aussi juste écrire `if "Contents" in page:`. C’est moins EAFP.

Reply:
Ça été ma première réflexion et puis je me suis dit que c'était une pagination donc il devait bien y avoir un moyen de tester qu'on a quelque chose :), mais vu que c'est déjà bizarre de ne pas inclure un `Content` vide alors peut-être que `KeyCount` c'est pareil :shrug:.
Mais plutôt +1 pour le `in` dans le cas présent alors.

Reply:
Allez, c’est parti pour le `in`. J’ai une petite préférence pour EAFP, mais vu le cleanup que ça demande... finalement `in` n’est pas si mal :)